app/src/org/commcare/utils/MediaUtil.java

```diff
@@ -370,21 +372,82 @@ private static Bitmap scaleDownToTargetOrContainer(String imageFilepath,
+            ExifInterface exif = new ExifInterface(imageFilepath);
+            orientation = exif.getAttributeInt(ExifInterface.TAG_ORIENTATION, ExifInterface.ORIENTATION_NORMAL);
+        } catch (IOException e) {
+            Log.e(TAG, "Unable to read EXIF data: " + e.getMessage());
+        }
+
+        // Rotate the bitmap if needed
+        Bitmap rotatedBitmap = rotateBitmap(b, orientation);
+        if (rotatedBitmap != b) {
+            b.recycle(); // Free up memory from the original bitmap
+        }
+
         if (scaleByContainerOnly && !respectBoundsExactly) {
             // Not worth performance loss of creating an exact scaled bitmap in this case
-            return b;
+            return rotatedBitmap;
         } else {
             try {
                 // Here we want to be more precise because we have a target width and height, or
                 // specified that respecting the bounding container precisely is important
-                return Bitmap.createScaledBitmap(b, newWidth, newHeight, false);
+                return Bitmap.createScaledBitmap(rotatedBitmap, newWidth, newHeight, false);
             } catch (OutOfMemoryError e) {
                 Log.d(TAG, "Ran out of memory attempting to scale image at: " + imageFilepath);
+                rotatedBitmap.recycle();
                 return null;
             }
         }
     }
 
+    // Helper method to rotate the bitmap based on EXIF orientation that we previously retained
+    private static Bitmap rotateBitmap(Bitmap bitmap, int orientation) {
+        if (bitmap == null || orientation == ExifInterface.ORIENTATION_NORMAL) {
+            return bitmap;
+        }
+
+        Matrix matrix = new Matrix();
+        switch (orientation) {
+            case ExifInterface.ORIENTATION_ROTATE_90:
+                matrix.postRotate(90);
+                break;
+            case ExifInterface.ORIENTATION_ROTATE_180:
+                matrix.postRotate(180);
+                break;
+            case ExifInterface.ORIENTATION_ROTATE_270:
+                matrix.postRotate(270);
+                break;
+            case ExifInterface.ORIENTATION_FLIP_HORIZONTAL:
+                matrix.postScale(-1, 1);
+                break;
+            case ExifInterface.ORIENTATION_FLIP_VERTICAL:
+                matrix.postScale(1, -1);
+                break;
+            case ExifInterface.ORIENTATION_TRANSPOSE:
+                matrix.postRotate(90);
+                matrix.postScale(-1, 1);
+                break;
+            case ExifInterface.ORIENTATION_TRANSVERSE:
+                matrix.postRotate(270);
+                matrix.postScale(-1, 1);
+                break;
+            default:
+                return bitmap;
+        }
+        try {
+            return Bitmap.createBitmap(bitmap, 0, 0, bitmap.getWidth(), bitmap.getHeight(), matrix, true);
+        } catch (OutOfMemoryError e) {
+            Log.d(TAG, "Ran out of memory attempting to rotate bitmap");

```

Comment:
would be great if we can use `Logger.exception` here and in the other `OutOfMemoryError` catch abvove in the calling method.